### PR TITLE
added 6D rotation transforms and helper

### PR DESCRIPTION
**Pull request by simon-schaefer**
- 6D rotation transforms to quaternions, rotation matrix, axis-angle
- angle axis to rotation matrix
- generalized inverse transformation

### CODE REVIEW

File: pytrafo/rotation_6d.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import kornia
+import torch
+from torch.nn import functional as F
+
+
+def rotation_6d_to_rotation_matrix(x):
```
Comment (dorianhenning) [inline pytrafo/rotation_6d.py:6]: typehints missing

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+torch
```
Comment (dorianhenning) [inline requirements.txt:1]: could we move that in the `setup.py`, I really prefer the `install_requires=...` parameter.

File: pytrafo/angle_axis.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import kornia
+import torch
+
+
+def angle_axis_to_matrix(x: torch.Tensor) -> torch.Tensor:
+    shape = x.shape[:-1]
```
Comment (dorianhenning) [inline pytrafo/angle_axis.py:6]: docstring missing

File: pytrafo/linalg.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+import torch
+
+
+def inverse_transformation(x: torch.Tensor) -> torch.Tensor:
+    """Inverse transformation tensor i.e. T_AB => T_BA for a arbitrary shaped transformation matrix.
+    
+    Args:
+        x: input transformation matrix (..., 4, 4).
```
Comment (dorianhenning) [inline pytrafo/linalg.py:11]: return type docstring

File: pytrafo/rotation_6d.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    rotmat = torch.stack((b1, b2, b3), dim=-1)
+    return rotmat.view(*x_shape[:-1], 3, 3)
+
+
+def rotation_6d_to_quaternion(x: torch.Tensor) -> torch.Tensor:
+    """Convert 6d rotation representation to quaternion representation.
+
+    Args:
+        x: 6d rotation tensor (..., 6).
+    """
```
Comment (dorianhenning) [inline pytrafo/rotation_6d.py:37]: return docstring

File: pytrafo/rotation_6d.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    return kornia.geometry.angle_axis_to_quaternion(x)
+
+
+def rotation_6d_to_axis_angle(x: torch.Tensor) -> torch.Tensor:
+    """Convert 6d rotation representation to axis angle (3D) representation.
+    https://stackoverflow.com/questions/12463487/obtain-rotation-axis-from-rotation-matrix-and-translation-vector-in-opencv
+
+    Args:
+        x: 6d rotation tensor (..., 6)
+    """
```
Comment (dorianhenning) [inline pytrafo/rotation_6d.py:48]: return docstring

File: pytrafo/rotation_6d.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    ax = (r[..., 2, 1] - r[..., 1, 2]) / norm * angle
+    ay = (r[..., 0, 2] - r[..., 2, 0]) / norm * angle
+    az = (r[..., 1, 0] - r[..., 0, 1]) / norm * angle
+    return torch.stack([ax, ay, az], dim=-1)
+
+
+def matrix_to_rotation_6d(x: torch.Tensor) -> torch.Tensor:
+    """Convert rotation matrix to 6d representation.
+    from https://pytorch3d.readthedocs.io/en/latest/_modules/pytorch3d/transforms/rotation_conversions.html
```
Comment (dorianhenning) [inline pytrafo/rotation_6d.py:66]: parameter and return docstring

File: pytrafo/rotation_6d.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    """
+    batch_dim = x.size()[:-2]
+    return x[..., :2].clone().reshape(batch_dim + (6,))
+
+
+def angle_axis_to_rotation_6d(x: torch.Tensor) -> torch.Tensor:
+    """Convert rotation in axis-angle representation to 6d representation."""
```
Comment (dorianhenning) [inline pytrafo/rotation_6d.py:73]: parameter and return docstring

**Review by dorianhenning — APPROVED** [review]
Nice work, some docstrings are missing and one suggestion regarding the `requirements.txt`. Thanks for the effort :)
